### PR TITLE
Request: Sort lines in .gitignore file

### PR DESCRIPTION
**Pull request by tkf**
I know that this is kind of a bikeshedding, but let me request to sort lines in .gitignore.  I think it's better since it makes generated file more deterministic.

### CODE REVIEW

File: src/generate.jl [hunk 1]
```diff
@@ -229,14 +229,11 @@ Returns an array of generated file/directory names.
...
     pkg = basename(pkg_dir)
-    seen = [".DS_Store"]
-    t.manifest || push!(seen, "/Manifest.toml")  # Only ignore manifests at the repo root.
-    patterns = vcat(map(p -> p.gitignore, values(t.plugins))...)
-    for pattern in patterns
-        if !in(pattern, seen)
-            push!(seen, pattern)
-        end
+    seen = mapfoldl(p -> p.gitignore, append!, values(t.plugins); init=[".DS_Store"])
+    if !t.manifest && "Manifest.toml" ∉ seen
```
Comment (tkf) [inline src/generate.jl]: `"Manifest.toml" ∉ seen` part is actually unrelated to the main request.  But it gets rid of redundant `/Manifest.toml` entry in `.gitignore` when `Manifest.toml` is already listed.

File: src/generate.jl [hunk 2]
```diff
@@ -229,14 +229,11 @@ Returns an array of generated file/directory names.
...
 function gen_gitignore(pkg_dir::AbstractString, t::Template)
     pkg = basename(pkg_dir)
-    seen = [".DS_Store"]
-    t.manifest || push!(seen, "/Manifest.toml")  # Only ignore manifests at the repo root.
-    patterns = vcat(map(p -> p.gitignore, values(t.plugins))...)
-    for pattern in patterns
-        if !in(pattern, seen)
-            push!(seen, pattern)
-        end
+    seen = mapfoldl(p -> p.gitignore, append!, values(t.plugins); init=[".DS_Store"])
```
Comment (christopher-dG) [inline src/generate.jl]: This can probably be renamed to `entries` now (more or less unrelated but imo it's more fitting).

**Comment by christopher-dG** [comment]
I like this change.  Just one thing, could you replace `∉` with `!in`? I generally try to avoid Unicode in package code as a personal preference.

**Comment by tkf** [comment]
Thanks for the code review. I included your suggestions in the patch.

**Comment by christopher-dG** [comment]
Lovely, thank you!
